pkg/util/util.go

```diff
@@ -10,6 +10,16 @@ import (
 	"github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
 )
 
+// Component represents type of jaeger component
+type Component string
+
+const (
```

Comment:
Not sure `util` is the right place for this. We have some constants already in the main `v1` package (github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1), but that might also not be suitable.

As all usages of this new constant are inside the `deployment` package, I'd keep it private there in a file named `const.go`. We can move it later if we need it somewhere else.

Reply:
I also use it in `account` package, so if I put it in `deployment` package I cannot use it in `account` (cyclic import).
Put it in `account` package? (there's `JaegerServiceAccountFor` func where it's used)

Reply:
Good idea, placing it close to the `JaegerServiceAccountFor` sounds logical to me.